reportportal_client/service.py

```diff
@@ -249,31 +250,55 @@ def finish_launch(self, end_time, status=None, **kwargs):
         logger.debug("finish_launch - ID: %s", self.launch_id)
         return _get_msg(r)
 
-    def get_launch_info(self):
+    def get_launch_info(self, max_retries=5):
         """Get the current launch information.
 
+        Perform "max_retries" attempts to get current launch information
+        with 0.5 second sleep between them.
+
+        :param int max_retries: Number of retries to get launch information.
         :return dict: launch information
         """
+        if self.launch_id is None:
+            return {}
+
         url = uri_join(self.base_url_v1, "launch/uuid", self.launch_id)
-        launch_info = _get_json(self.session.get(
-            url=url, verify=self.verify_ssl))
-        logger.debug("get_launch_info - ID: %s", self.launch_id)
-        logger.debug("get_launch_info - Launch info: %s", launch_info)
+
+        for _ in range(max_retries):
+            logger.debug("get_launch_info - ID: %s", self.launch_id)
+            resp = self.session.get(url=url, verify=self.verify_ssl)
+
+            if resp.status_code == 200:
+                launch_info = _get_json(resp)
+                logger.debug("get_launch_info - Launch info: %s", launch_info)
+                break
+
+            logger.debug("get_launch_info - Launch info: Response code %s\n%s",
+                         resp.status_code, resp.text)
+            sleep(0.5)
+        else:
+            logger.warning("get_launch_info - Launch info: "
+                           "Failed to fetch launch ID from the API.")
+            launch_info = {}
+
         return launch_info
 
-    def get_launch_ui_id(self):
+    def get_launch_ui_id(self, max_retries=5):
         """Get UI ID of the current launch.
 
         :return str: UI ID of the given launch.
+                     0 if UI ID has not been found.
         """
-        return self.get_launch_info()["id"]
+        return self.get_launch_info(max_retries=max_retries).get("id", 0)
```

Comment:
Hi, 

Thanks a lot for the timely review! Sorry, I`m delaying changes sometimes. The purpose of `0` instead of `None` - is to allow a function to return values of the same type. This can prevent some bugs and extra code on the caller side. e.g. <> with `None`. Not a big deal here. WIll remove.